src/packages/solid/utils/getWidthByColumnSpan.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2024 Comcast Cable Communications Management, LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import theme from 'theme';
+import { getWidthByUpCount } from './getWidthByUpCount.js';
+export function getWidthByColumnSpan(columnSpan: number) {
+  const columnCount = theme.layout.columnCount;
+  const gutterX = theme.layout.gutterX;
+
+  return (getWidthByUpCount(columnCount) ?? 1) * columnSpan + gutterX * (columnSpan - 1);
```

Comment:
do we need the `??` here? getWidthByUpCount should only ever return numeric values

Reply:
I agree, but I get a possibly undefined ts error so I wasnt sure